homeassistant/components/bayesian/binary_sensor.py

```diff
@@ -78,6 +80,55 @@
     required=True,
 )
 
+
+def _above_greater_than_below(configs: list[dict[str, Any]]) -> list:
+    for config in configs:
+        if config.get(CONF_PLATFORM, None) != CONF_NUMERIC_STATE:
+            continue
+        above = config.get(CONF_ABOVE, None)
+        below = config.get(CONF_BELOW, None)
+        if above is None and below is None:
+            raise vol.Invalid(
+                "For bayesian numeric state at least one of 'above' or 'below' must be specified."
+            )
+        if above is not None and below is not None:
+            if above > below:
+                raise vol.Invalid(
+                    f"For bayesian numeric state 'above' ({above}) must be less than 'below' ({below})."
+                )
+    return configs
+
+
+def _no_overlaping(configs: list[dict]) -> list[dict]:
+    numeric_configs = [
+        config
+        for config in configs
+        if config.get(CONF_PLATFORM, None) == CONF_NUMERIC_STATE
+    ]
+    if len(numeric_configs) < 2:
+        return configs
+    Numeric_config = namedtuple("Numeric_config", ["above", "below"])
+    d: dict[str, list[Numeric_config]] = {}
+    for _, config in enumerate(numeric_configs):
+        above = config.get(CONF_ABOVE, -math.inf)
+        below = config.get(CONF_BELOW, math.inf)
+        entity_id: str = str(config.get(CONF_ENTITY_ID))
+        d.setdefault(entity_id, [])
+        d[entity_id].append(Numeric_config(above, below))
+
+    for ent_id, named_tuples in d.items():
```

Comment:
named_tuple is a bad name... what about `interval` ?

Reply:
changed to `intervals`